stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -169,60 +178,168 @@ public static String getErrorMessage(Throwable err) {
-                                                                            WebSocketConstants.WEBSOCKET_ERROR_DETAILS);
-        return BallerinaValues.createRecord(detail, errMsg, cause);
+    private static void waitForHandshake(ObjectValue webSocketClient, CountDownLatch countDownLatch) {
+
+        long timeout = WebSocketUtil.findTimeoutInSeconds((MapValue<String, Object>) webSocketClient.getMapValue(
+                HttpConstants.CLIENT_ENDPOINT_CONFIG), "handShakeTimeoutInSeconds", 300);
+        try {
+            if (!countDownLatch.await(timeout, TimeUnit.SECONDS)) {
+                throw new WebSocketException(WebSocketConstants.ErrorCode.WsGenericError,
+                        "Waiting for WebSocket handshake has not been successful", WebSocketUtil.createErrorCause(
+                        "Connection timeout", IOConstants.ErrorCode.ConnectionTimedOut.errorCode(),
+                        IOConstants.IO_PACKAGE_ID));
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new WebSocketException(WebSocketConstants.ERROR_MESSAGE + e.getMessage());
+        }
+    }
+
+    /**
+     * Reconnect when the WebSocket connection is lost.
+     *
+     * @param connectionInfo - information about the connection.
+     * @return If attempts reconnection, then return true.
+     */
+    public static boolean reconnect(WebSocketConnectionInfo connectionInfo) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        RetryContext retryConnectorConfig = (RetryContext) webSocketClient.getNativeData(WebSocketConstants.
+                RETRY_CONFIG);
+        int interval = retryConnectorConfig.getInterval();
+        int maxInterval = retryConnectorConfig.getMaxInterval();
+        int maxAttempts = retryConnectorConfig.getMaxAttempts();
+        int noOfReconnectAttempts = retryConnectorConfig.getReconnectAttempts();
+        float backOfFactor = retryConnectorConfig.getBackOfFactor();
+        WebSocketService wsService = connectionInfo.getService();
+        Date date = new Date();
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+        // Check if the no of reconnection attempts is equal to the maximum number of reconnection attempts or not.
+        // If it is not equal, then call the `initialiseWebSocketConnection()` function.
+        // If it is equal, then return false.
+        if (((noOfReconnectAttempts < maxAttempts) && maxAttempts > 0) || maxAttempts == 0) {
+            retryConnectorConfig.setReconnectAttempts(noOfReconnectAttempts + 1);
+            String time = formatter.format(date.getTime());
+            logger.debug(WebSocketConstants.LOG_MESSAGE, time, "reconnecting...");
+            setCountDownLatch(calculateWaitingTime(interval, maxInterval, backOfFactor, noOfReconnectAttempts));
+            establishWebSocketConnection(webSocketClient, wsService);
+            return true;
```

Comment:
I don't see any use for the boolean return value? Why do we have it?

Reply:
It is using following places

https://github.com/ballerina-platform/ballerina-lang/pull/19723/files#diff-d98104e28f7ae983c0ed1ac50690dde5R80

https://github.com/ballerina-platform/ballerina-lang/pull/19723/files#diff-c344ae55fd805ed9fdc231708f0e342fR113